### ISSUE TITLE
[Chore] Error management 

### ISSUE THREAD

**Issue by Swiizyy** [BirthdayyBot/BirthdayyBotSapphire#191]
Currently we use a mix between "try/catch" and "await/then/catch".

I think it would be better to choose one way among these two and to use only this one in order not to lose and to manage it better.

```ts
// try/catch
export async function getGuildInformation(guildId: string): Promise<Guild | null> {
	try {
		return container.client.guilds.fetch(guildId);
	} catch (error) {
		if (error instanceof Error) {
			container.logger.error(`Error fetching guild with id ${guildId}:`, error.message);
		}
		return null;
	}
}

// await/catch
export async function getGuildInformation(guildId: string): Promise<Guild | null> {
	return container.client.guilds.fetch(guildId).catch( (error) => {
		if (error instanceof Error) container.logger.error(`Error fetching guild with id ${guildId}:`, error.message);
		return null;
	}
}
```

In our case I think it would be better to use "await/then/catch" for more simplicity and understanding of the code.

@nikolaischunk  What do you think ?

**Comment by nikolaischunk** [comment]
I agree with you, the second statement does indeed look cleaner.
Good Idea for a Refactor

**Comment by Swiizyy** [comment]
> I agree with you, the second statement does indeed look cleaner.
> 
> Good Idea for a Refactor

Yes is then try/catch should be used on synchronous and non-asynchronous functions 🙂

**Comment by Swiizyy** [comment]
Currently we manage errors in a more than rough way. It would be good to set up better error management to facilitate this one.

We can handle sending an error to sentry in different listeners that will be activated only if SENTRY_URL is defined.

Send only useful errors to sentry. Example: User/Guild/Channel fetch error is not useful

**Comment by Swiizyy** [comment]
Currently, all the classes we use have all the logic in one function "run" it would be better to separate this logic into several functions and call them with "this." for more maintainability

**Comment by nikolaischunk** [comment]
it's only cluttered because I activated `DEBUG` in PRD. 
logging many things for debugging purposes is valuable and should still be done.

Reporting errors like `User/Guild/Channel fetch error is not useful` is done so that we know something went wrong.
I agree this doesn't have to be added to the Sentry, but catching the error and doing something with it(for example deleting the guild entry in the database) still makes sense.
